### PR TITLE
Crypto: improve PSP and PSVita PKG interpretation

### PR DESCRIPTION
**Pull request by Megamouse**
also enables installation with one more filetype

### CODE REVIEW

File: rpcs3/Crypto/unpkg.h [hunk 1]
```diff
@@ -47,6 +48,23 @@ struct PKGHeader
 	char title_id[48];      // Title ID
 	be_t<u64> qa_digest[2]; // This should be the hash of "files + attribs"
 	be_t<u128> klicensee;   // Nonce
+	// + some stuff
+};
+
+// Extended header in PSP and PSVita packages
+struct PKGExtHeader
+{
+	nse_t<u32> magic;                            // 0x7F657874 (" ext")
```
Comment (elad335) [inline rpcs3/Crypto/unpkg.h:57]:
```suggestion
	le_t<u32> magic;                            // 0x7F657874 (" ext")
```

File: rpcs3/Crypto/unpkg.h [hunk 2]
```diff
@@ -34,7 +35,7 @@ enum : u32
 // Structs
 struct PKGHeader
 {
-	nse_t<u32> pkg_magic;   // Magic (0x7f504b47)
+	nse_t<u32> pkg_magic;   // Magic (0x7f504b47) (" PKG")
```
Comment (elad335) [inline rpcs3/Crypto/unpkg.h:38]:
```suggestion
	le_t<u32> pkg_magic;   // Magic (0x7f504b47) (" PKG")
```

File: rpcs3/Crypto/unpkg.cpp [hunk 3]
```diff
@@ -78,10 +85,50 @@ bool pkg_install(const std::string& path, atomic_t<double>& sync)
...
+		pkg_log.notice("Extended header: main_and_ext_headers_hmac_offset = 0x%x = %d", ext_header.main_and_ext_headers_hmac_offset, ext_header.main_and_ext_headers_hmac_offset);
+		pkg_log.notice("Extended header: metadata_header_hmac_offset = 0x%x = %d", ext_header.metadata_header_hmac_offset, ext_header.metadata_header_hmac_offset);
+		pkg_log.notice("Extended header: tail_offset = 0x%x = %d", ext_header.tail_offset, ext_header.tail_offset);
+		//pkg_log.notice("Extended header: padding1 = 0x%x = %d", ext_header.padding1, ext_header.padding1);
+		pkg_log.notice("Extended header: pkg_key_id = 0x%x = %d", ext_header.pkg_key_id, ext_header.pkg_key_id);
+		pkg_log.notice("Extended header: full_header_hmac_offset = 0x%x = %d", ext_header.full_header_hmac_offset, ext_header.full_header_hmac_offset);
+		//pkg_log.notice("Extended header: padding2 = 0x%x = %d", ext_header.padding2, ext_header.padding2);
+	}
+
 	if (header.pkg_magic != "\x7FPKG"_u32)
```
Comment (elad335) [inline rpcs3/Crypto/unpkg.cpp:132]:
```suggestion
	if (header.pkg_magic != std::bit_cast<le_t<u32>>("\x7FPKG"_u32))
```

File: rpcs3/Crypto/unpkg.h (hunk 2 above)
Comment (elad335) [inline rpcs3/Crypto/unpkg.h:38]:
```suggestion
	le_t<u32> pkg_magic;   // Magic (0x7f504b47) (" PKG")
```

File: rpcs3/Crypto/unpkg.h (hunk 1 above)
Comment (elad335) [inline rpcs3/Crypto/unpkg.h:57]:
```suggestion
	le_t<u32> magic;                            // 0x7F657874 (" ext")
```

**Comment by StalkerDolg** [comment]
This is for what? Its a PS3 emulator. Or RPCS3 will be emulate a psp and vita?

**Comment by AniLeo** [comment]
O really that it's a PS3 emulator? There is shared PSV + PS3 content and many PSP packages for PS3 which runs PSP games

**Comment by StalkerDolg** [comment]
> O really that it's a PS3 emulator? There is shared PSV + PS3 content and many PSP packages for PS3 which runs PSP games

I know it offtop, but. RPCS3 team planning to add a quick saves to emulator?

**Comment by AniLeo** [comment]
Packages and savedata have nothing to do with eachother.

**Pull request merged**